pkg/operation/botanist/controlplane/kubescheduler/kube_scheduler.go

```diff
@@ -299,13 +311,80 @@ func (k *kubeScheduler) emptyVPA() *autoscalingv1beta2.VerticalPodAutoscaler {
 }
 
 func (k *kubeScheduler) emptyService() *corev1.Service {
-	return &corev1.Service{ObjectMeta: metav1.ObjectMeta{Name: "kube-scheduler", Namespace: k.namespace}}
+	return &corev1.Service{ObjectMeta: metav1.ObjectMeta{Name: ServiceName, Namespace: k.namespace}}
 }
 
 func (k *kubeScheduler) emptyDeployment() *appsv1.Deployment {
 	return &appsv1.Deployment{ObjectMeta: metav1.ObjectMeta{Name: v1beta1constants.DeploymentNameKubeScheduler, Namespace: k.namespace}}
 }
 
+func (k *kubeScheduler) emptyManagedResource() *resourcesv1alpha1.ManagedResource {
+	return &resourcesv1alpha1.ManagedResource{ObjectMeta: metav1.ObjectMeta{Name: managedResourceName, Namespace: k.namespace}}
+}
+
+func (k *kubeScheduler) emptyManagedResourceSecret() *corev1.Secret {
+	secretName, _ := common.NewManagedResourceSecret(k.client, managedResourceName, k.namespace, nil)
```

Comment:
Can you rather extract the name computation logic and use a dedicated function for that?